domain/user/bootstrap/bootstrap.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package bootstrap
+
+import (
+	"context"
+
+	"github.com/canonical/sqlair"
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/core/database"
+	"github.com/juju/juju/core/user"
+	"github.com/juju/juju/domain/user/state"
+)
+
+const adminUserName = "admin"
+const adminUserDisplayName = "admin"
+
+// GenerateAdminUser inserts the admin user into database.
+// It is used to bootstrap the database.
+//
+// Admin user is created with the following characteristics:
+// 1. This is first user created in the system.
+// 2. This user is used to owner of the first model created in the system.
+// 3. This user is created with no password authorization by default.
+func GenerateAdminUser() func(context.Context, database.TxnRunner) error {
```

Comment:
Apologies @anvial I should have had a read of the bootstrap code yesterday to guide you on this one a bit more.

So this func is correct but needs a few changes. If you look at agent bootstrap as part of the state initialisation params the client passes us an admin user tag that has the username of the admin user in it. It also passes us the password for the admin user.

So all this bootstrap func needs to do is mimic the signature and behaviour of AddUserWithPassword in the services layer and also return the uuid so we can use it as input into other places. Hope that makes sense but happy to discuss it a bit more with you.